vsintegration/src/FSharp.ProjectSystem.Base/Project/ProjectFactory.cs

```diff
@@ -146,6 +203,67 @@ protected override object PreCreateForOuter(IntPtr outerProjectIUnknown)
+        {
+            // Solution properties
+            IVsSolution solution = this.Site.GetService(typeof(SVsSolution)) as IVsSolution;
+            Debug.Assert(solution != null, "Could not retrieve the solution service from the global service provider");
+
+            // We do not want to throw.If we cannot set the solution related constants we set them to empty string.
+            string solutionDirectory, solutionFile, userOptionsFile;
+            solution.GetSolutionInfo(out solutionDirectory, out solutionFile, out userOptionsFile);
+            if (solutionDirectory == null)
+            {
+                solutionDirectory = String.Empty;
+            }
+            if (solutionFile == null)
+            {
+                solutionFile = String.Empty;
+            }
+            string solutionFileName = (solutionFile.Length == 0) ? String.Empty : Path.GetFileName(solutionFile);
+            string solutionName = (solutionFile.Length == 0) ? String.Empty : Path.GetFileNameWithoutExtension(solutionFile);
+            string solutionExtension = String.Empty;
+            if (solutionFile.Length > 0 && Path.HasExtension(solutionFile))
+            {
+                solutionExtension = Path.GetExtension(solutionFile);
+            }
+
+            //DevEnvDir property
+            IVsShell shell = this.Site.GetService(typeof(SVsShell)) as IVsShell;
+            Debug.Assert(shell != null, "Could not retrieve the IVsShell service from the global service provider");
+            object installDirAsObject;
+
+            //We do not want to throw.If we cannot set the solution related constants we set them to empty string.
+            shell.GetProperty((int)__VSSPROPID.VSSPROPID_InstallDirectory, out installDirAsObject);
+            string installDir = ((string)installDirAsObject);
+            if (String.IsNullOrEmpty(installDir))
+            {
+                installDir = String.Empty;
+            }
+            else
+            {
+                //Ensure that we have traimnling backslash as this is done for the langproj macros too.
+                if (installDir[installDir.Length - 1] != Path.DirectorySeparatorChar)
+                {
+                    installDir += Path.DirectorySeparatorChar;
+                }
+            }
+
+            var projectGlobalPropertiesThatAllProjectSystemsMustSet = new Dictionary<string, string>()
+            {
+                { GlobalProperty.SolutionDir.ToString(), solutionDirectory },
+                { GlobalProperty.SolutionPath.ToString(), solutionFile },
+                { GlobalProperty.SolutionFileName.ToString(), solutionFileName },
+                { GlobalProperty.SolutionName.ToString(), solutionName },
+                { GlobalProperty.SolutionExt.ToString(), solutionExtension },
+                { GlobalProperty.BuildingInsideVisualStudio.ToString(), "true" },
+                { GlobalProperty.Configuration.ToString(), "" },
```

Comment:
Are there ramifications to setting Configuration and Platform to empty?  Or at least, can you re-check that switching configuration/platform works correctly on common paths, and get set to the right values on solution load/reload/edit/...

Reply:
So @dsyme ... up until this change they were never set on these code paths at all, and so would be blank until configured in the project.  In this case the setting is just .. completeness rather than anything else.